src/main/java/com/eu/habbo/habbohotel/rooms/Room.java

```diff
@@ -4873,4 +4877,162 @@ public Collection<RoomUnit> getRoomUnitsAt(RoomTile tile) {
+                                furniOwnerNames.put(item.getUserId(), habbo.getUsername());
+                            }
+                        }
+                    }
+
+                    if (data.on == 1) {
+                        if (!floorItems.isEmpty()) {
+                            this.sendComposer(new ObjectRemoveMultipleComposer(floorItems).compose());
+                        }
+                        if (!wallItems.isEmpty()) {
+                            wallItems.stream().forEach(item -> this.sendComposer(new ItemRemoveComposer(item, true).compose()));
+                        }
+                    } else {
+                        if (!floorItems.isEmpty()) {
+                            this.sendComposer(new ObjectsComposer(furniOwnerNames, floorItems).compose());
+                        }
+                        if (!wallItems.isEmpty()) {
+                            this.sendComposer(new ItemsComposer(this).compose());
+                        }
+                    }
+
+                } catch (JsonSyntaxException e) {
+                    LOGGER.error("Error upon parsing extradata of AreaHider({}), error: {}", areaHider.getId(), e.getMessage());
+                }
+            }
+        }
+    }
+
+    public Map<String, THashSet<HabboItem>> getItemsInRectangle(Rectangle rectangle, boolean invert, boolean wallitems) {
+        TIntObjectIterator<HabboItem> iterator = this.roomItems.iterator();
+        THashSet<HabboItem> foundFloorItems = new THashSet<>();
+        THashSet<HabboItem> foundWallItems = new THashSet<>();
+
+        while (iterator.hasNext()) {
+            try {
+                iterator.advance();
+                HabboItem object = iterator.value();
+
+                if (object instanceof InteractionAreaHider) {
+                    continue;
+                }
+
+                short x = object.getX();
+                short y = object.getY();
+
+                if (object.getBaseItem().getType() == FurnitureType.WALL) {
+                    x += (short) ((object.getRotation() == 0) ? 1 : 0);
+                    y += (short) ((object.getRotation() == 1) ? 1 : 0);
+                }
+
+                RoomTile tile = this.getLayout().getTile(x, y);
+                boolean isInRectangle = RoomLayout.tileInSquare(rectangle, tile);
+
+                if (object.getBaseItem().getType() == FurnitureType.FLOOR && ((invert && !isInRectangle) || (!invert && isInRectangle))) {
+                    foundFloorItems.add(object);
+                }
+                if (wallitems && object.getBaseItem().getType() == FurnitureType.WALL && ((invert && !isInRectangle) || (!invert && isInRectangle))) {
+                    foundWallItems.add(object);
+                }
+            } catch (Exception e) {
+                LOGGER.error("An error occurred: ", e);
+            }
+        }
+
+        Map<String, THashSet<HabboItem>> resultMap = new HashMap<>();
```

Comment:
Make this a `class` (or `record` if we ditch java 8) with two fields instead of a hashmap with hardcoded fields.